src/ImageStats/BasicStatsCalculator.h

```diff
@@ -32,10 +32,10 @@ class BasicStatsCalculator {
     T _min_val, _max_val;
     double _sum, _sum_squares;
     size_t _num_pixels;
-    const std::vector<T>& _data;
+    const float* _data;
 
 public:
-    BasicStatsCalculator(const std::vector<T>& data);
+    BasicStatsCalculator(const float* data, size_t data_size);
```

Comment:
Yes, you are right, I might have automatically included the variable when moving to the pointer + size design, but it is unneeded here. I will drop what is unneeded here for the change, but we should file a new issue for improving the design of the class if we want to.

Reply:
To resolve this: I included `data_size` as a member and added a small range check to `BasicStatsCalculator::reduce`. I think it's ok doing so. I also realized we must use `const T*` instead of `const float*` since this is a template class.